api/api_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+ *    Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package api
+
+import (
+	"testing"
+
+	"github.com/Peripli/service-manager/rest"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestApi(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "API Suite")
+}
+
+type testController struct {
+}
+
+func (c *testController) Routes() []rest.Route {
+	return []rest.Route{}
+}
+
+var _ = Describe("API", func() {
+	var defaultAPI rest.API
+
+	BeforeEach(func() {
+		defaultAPI = Default()
+	})
+
+	Describe("Controller Registration", func() {
+		JustBeforeEach(func() {
+			defaultAPI.RegisterControllers(&testController{})
+		})
+		Context("With nil controllers slice", func() {
+			It("Should panic", func() {
+				nilControllersSlice := func() {
+					defaultAPI.RegisterControllers(nil)
+				}
+				Expect(nilControllersSlice).To(Panic())
+			})
+		})
+
+		Context("With nil controller in slice", func() {
+			It("Should panic", func() {
+				nilControllerInSlice := func() {
+					var controllers []rest.Controller
+					controllers = append(controllers, nil)
+					defaultAPI.RegisterControllers(controllers...)
+				}
+				Expect(nilControllerInSlice).To(Panic())
+			})
+		})
+
+		Context("With no brokers registered", func() {
+			It("Should have only one broker", func() {
+				Expect(len(defaultAPI.Controllers())).To(Equal(1))
```

Comment:
I'd suggest to not check for hard-coded values in tests (like in this case that exactly 1 is present) - for example it might be cleaner to get the count in the begining of the test, add one item, then get the new count and validate that it has increased by one.

You may look at https://semaphoreci.com/community/tutorials/getting-started-with-bdd-in-go-using-ginkgo and search for `originalItemCount`